adobe_vipm/flows/fulfillment/purchase.py

```diff
@@ -32,302 +31,243 @@
     STATUS_MARKET_SEGMENT_PENDING,
     TEMPLATE_NAME_PURCHASE,
 )
+from adobe_vipm.flows.context import Context
 from adobe_vipm.flows.fulfillment.shared import (
-    add_subscription,
-    check_adobe_order_fulfilled,
-    check_processing_template,
-    get_one_time_skus,
-    save_adobe_customer_data,
-    save_adobe_order_id,
-    save_next_sync_and_coterm_dates,
-    switch_order_to_completed,
+    CompleteOrder,
+    CreateOrUpdateSubscriptions,
+    IncrementAttemptsCounter,
+    SetOrUpdateCotermNextSyncDates,
+    StartOrderProcessing,
+    SubmitNewOrder,
+    UpdatePrices,
+    ValidateDuplicateLines,
     switch_order_to_failed,
     switch_order_to_query,
-    update_order_actual_price,
 )
-from adobe_vipm.flows.helpers import prepare_customer_data
+from adobe_vipm.flows.helpers import SetupContext, prepare_customer_data
+from adobe_vipm.flows.mpt import update_agreement, update_order
+from adobe_vipm.flows.pipeline import Pipeline, Step
 from adobe_vipm.flows.utils import (
-    get_adobe_customer_id,
-    get_adobe_order_id,
-    get_market_segment,
     get_market_segment_eligibility_status,
     get_ordering_parameter,
+    set_adobe_3yc_commitment_request_status,
+    set_adobe_customer_id,
     set_market_segment_eligibility_status_pending,
     set_order_error,
     set_ordering_parameter_error,
 )
-from adobe_vipm.utils import get_partial_sku
 
 logger = logging.getLogger(__name__)
 
 
-def _handle_customer_error(client, order, error):
+class RefreshCustomer(Step):
     """
-    Processes the error received from the Adobe API during customer creation.
-    If the error is related to a customer parameter, the parameter error attribute
-    is set, and the MPT order is switched to the 'query' status.
-    Other errors will result in the MPT order being marked as failed.
+    Refresh the processing context
+    retrieving the Adobe customer object through the
+    VIPM API.
+    """
+
+    def __call__(self, client, context, next_step):
+        adobe_client = get_adobe_client()
+        context.adobe_customer = adobe_client.get_customer(
+            context.authorization_id,
+            context.adobe_customer_id,
+        )
+        next_step(client, context)
```

Comment:
BTW, I was thinking about it. Do you think that using a return statement is better to control the flow? In this case, the Pipeline code itself should switch to the next step. What is needed from the step code is to return updated context. Otherwise we should be strict here to always call `next_step`

Reply:
next_step will be called from the current step only if the current_step completed successfully. For example the SubmitNewOrder step only go to the next one when Adobe reports the order has been processed. Otherwise it stop the processing. IMO returning or not returning the context to make the pipeline understand if it has to continue or stop is more or less the same, probably a bit less clear.